unified_planning/io/anml_reader.py

```diff
@@ -0,0 +1,810 @@
+                                        f"Currently the UP does not support the parsing of the {operator} operator."
+                                    )
+                        else:
+                            raise NotImplementedError(
+                                f"Currently the UP does not support the expression {exp}"
+                            )
+                else:  # expression longer than 3, must be a parameters list
+                    for e in exp:
+                        assert isinstance(e, List) or isinstance(
+                            e, ParseResults
+                        ), f"expression {exp} is expected to be a parameters list, but it's not"
+                        pass
+            else:  # not solved
+                if isinstance(exp, ParseResults) or isinstance(exp, List):
+                    first_token = exp[0]
+                    if isinstance(first_token, str) and first_token in (
+                        TK_FORALL,
+                        TK_EXISTS,
+                    ):  # quantifier
+                        assert isinstance(exp, ParseResults)
+                        name_type = self._parse_parameters_def(
+                            exp["quantifier_variables"], types_map
+                        )
+                        new_vars = {n: Variable(n, t) for n, t in name_type.items()}
+                        stack.append((exp, True, new_vars))
+                        all_vars = vars.copy()
+                        all_vars.update(new_vars)
+                        for e in exp["quantifier_body"]:
+                            stack.append((e, False, all_vars))
+                    else:
+                        stack.append((exp, True, vars))
+                        for e in exp:
+                            if not isinstance(e, str):  # nested structure
+                                if len(e) > 0:
+                                    stack.append((e, False, vars))
+                            elif e.isnumeric():  # int
+                                solved.append(self._em.Int(int(e)))
+                            elif is_float(e):  # float
+                                solved.append(self._em.Real(Fraction(float(e))))
+                            elif e == TK_TRUE:  # true
+                                solved.append(self._em.TRUE())
+                            elif e == TK_FALSE:  # false
+                                solved.append(self._em.FALSE())
+                elif isinstance(exp, str):
+                    assert False, "problem, should not have strings here"
+                else:
+                    raise NotImplementedError
+        assert len(stack) == 0
+        assert len(solved) == 1, f"{expression}"
+        res = solved.pop()
+        return res.simplify()
+
+
+def is_float(string: str) -> bool:
+    try:
+        float(string)
+        return True
+    except ValueError:
+        return False
+
+
+def count_instances(
```

Comment:
Do you really need the exact number of occurrences?

Reply:
Yes, sometimes it is useful to validate the input without having to re-check all the parsed result.